crates/router/src/core/webhooks/utils.rs

```diff
@@ -154,3 +158,98 @@ pub(crate) fn get_idempotent_event_id(
 pub(crate) fn generate_event_id() -> String {
     common_utils::generate_time_ordered_id("evt")
 }
+
+pub async fn perform_redis_lock<A>(
+    state: &A,
+    unique_locking_key: String,
+    merchant_id: common_utils::id_type::MerchantId,
+) -> RouterResult<bool>
+where
+    A: SessionStateInfo,
+{
+    let redis_conn = state
+        .store()
+        .get_redis_conn()
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+
+    let redis_locking_key = format!(
+        "{}_{}_{}",
+        WEBHOOK_LOCK_PREFIX,
+        merchant_id.get_string_repr(),
+        unique_locking_key
+    );
+    let redis_lock_expiry_seconds = state.conf().lock_settings.redis_lock_expiry_seconds;
```

Comment:
@jarnura Are we okay with using the same TTL as that used for API locking?

Reply:
Have a separate locking expiry time